docs/federate.md

```diff
@@ -0,0 +1,156 @@
+Setting up Federation
+=====================
+
+Federation is the process by which users on different servers can participate
+in the same room. For this to work, those other servers must be able to contact
+yours to send messages.
+
+The ``server_name`` configured in the Synapse configuration file (often
+``homeserver.yaml``) defines how resources (users, rooms,...) will be
+identified (ie: ``@user:example.com``, ``#room:example.com``). By
+default, it is also the default domain that other servers will use to
+try to reach your server via port 8448. This is easy to set
+up and will work provided you set the ``server_name`` to match your
+machine's public DNS hostname, and provide Synapse with a TLS certificate
+which is valid for your ``server_name``.
+
+For a more flexible configuration, you can have ``server_name``
+resources (ie: ``@user:example.com``) served by a different host and
+port (ie: ``synapse.example.com:443``). There are 2 ways to do this:
+
+- adding a DNS ``SRV`` record in the DNS zone of domain
+  ``example.com``. Beware that this method has some limitations as it
+  will still require your delegated server to use a SSL certificate
+  identifying it as the original ``server_name`` domain name. Meaning
+  that the provided ``synapse.example.com`` delegate domain name will
+  only be used to get a possibly different IP/port, but won't be used
+  for SSL domain name verification.
+
+- adding a ``/.well-known/matrix/server`` URL served on ``https://example.com``
+
+For both methods let's say you want to run your server at
```

Comment:
these three paragraphs are probably redundant?

Reply:
I see it as more of an intro - I've removed quite a lot as per other comments and added a Delegation section. I think it is in better shape. I'll push so you can see.

Reply:
I still think this paragraph is just duplicating the paragraph above "For a more flexible configuration ...".